src/module.c

```diff
@@ -3379,6 +3379,40 @@ int RM_GetClientInfoById(void *ci, uint64_t id) {
     return modulePopulateClientInfoStructure(ci,client,structver);
 }
 
+/* Returns the name of the client connection with the given ID.
+ *
+ * If the client ID does not exist or if the client has no name associated with
+ * it, NULL is returned. */
+RedisModuleString *RM_GetClientNameById(RedisModuleCtx *ctx, uint64_t id) {
+    client *client = lookupClientByID(id);
+    if (client == NULL || client->name == NULL) return NULL;
+    robj *name = client->name;
+    incrRefCount(name);
+    autoMemoryAdd(ctx, REDISMODULE_AM_STRING, name);
+    return name;
+}
+
+/* Sets the name of the client with the given ID. This is equivalent to the client calling
+ * `CLIENT SETNAME name`.
+ *
+ * Returns REDISMODULE_OK on success. On failure, REDISMODULE_ERR is returned
+ * and errno is set as follows:
+ *
+ * - ENOENT if the client does not exist
+ * - EINVAL if the name contains invalid characters */
+int RM_SetClientNameById(uint64_t id, RedisModuleString *name) {
```

Comment:
i'm still not certain we wanna allow changing a client name for an arbitrary client id

Reply:
TBH, I don't have a use case for any of this. The idea was to make it possible to intercept HELLO to allow username/password handling, but if we solve that in another way, I'm not sure anyone has a use case for this. Perhaps I can just close this PR?

Reply:
i think at the very least we do need to let the module read the client name (and local address).
so maybe we should expose that in the other PR i suggested to start.

Reply:
Yes. I'll look into REDISMODULE_CLIENTINFO_INITIALIZER and the event version problem.